packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts

```diff
@@ -0,0 +1,130 @@
+            !erc721TokenContract ||
+            !contractDetailed?.address ||
+            !EthereumAddress.isValid(contractDetailed?.address) ||
+            !owner
+        )
+            return
+
+        let lists: ERC721TokenDetailed[]
+
+        if (!GET_ASSETS_URL) {
+            lists = await getERC721TokenDetailedOwnerListFromChain(erc721TokenContract, contractDetailed, owner, offset)
+            allListRef.current = allListRef.current.concat(lists)
+            return { tokenDetailedOwnerList: allListRef.current, loadMore: lists.length > 0 }
+        }
+
+        const tokenDetailedOwnerListFromOpensea = await getERC721TokenDetailedOwnerListFromOpensea(
+            contractDetailed,
+            owner,
+            GET_ASSETS_URL,
+            offset,
+        )
+
+        lists =
+            tokenDetailedOwnerListFromOpensea ??
+            (await getERC721TokenDetailedOwnerListFromChain(erc721TokenContract, contractDetailed, owner, offset))
+
+        allListRef.current = allListRef.current.concat(lists)
+
+        return { tokenDetailedOwnerList: allListRef.current, loadMore: lists.length > 0 }
+    }, [GET_ASSETS_URL, contractDetailed, owner, offset])
+    const clearTokenDetailedOwnerList = () => (allListRef.current = [])
+    return { asyncRetry, clearTokenDetailedOwnerList }
+}
+
+async function getERC721TokenDetailedOwnerListFromChain(
+    erc721TokenContract: ERC721,
+    contractDetailed: ERC721ContractDetailed,
+    owner: string,
+    offset: number,
+) {
+    const queryLimit = 10
+    const isEnumerable = await safeNonPayableTransactionCall(
+        erc721TokenContract.methods.supportsInterface(ERC721_ENUMERABLE_INTERFACE_ID),
+    )
+
+    const balance = await safeNonPayableTransactionCall(erc721TokenContract.methods.balanceOf(owner))
+
+    if (!isEnumerable || !balance) return []
+
+    const allRequest = Array.from({ length: min([Number(balance), queryLimit])! }).map(async (_v, i) => {
+        const tokenId = await safeNonPayableTransactionCall(
+            erc721TokenContract.methods.tokenOfOwnerByIndex(owner, i + offset * queryLimit),
+        )
+
+        if (tokenId) {
+            const tokenDetailed = await getERC721TokenDetailedFromChain(contractDetailed, erc721TokenContract, tokenId)
+            return tokenDetailed
+        }
+
+        return
+    })
+
+    const tokenDetailedOwnerList = (await Promise.allSettled(allRequest))
+        .map((x) => (x.status === 'fulfilled' ? x.value : undefined))
+        .filter((value) => value) as ERC721TokenDetailed[]
+
+    return tokenDetailedOwnerList
+}
+
+async function getERC721TokenDetailedOwnerListFromOpensea(
+    contractDetailed: ERC721ContractDetailed,
+    owner: string,
+    apiURL: string,
+    offset: number,
+) {
+    const response = await fetch(
+        urlcat(apiURL, {
+            owner,
+            asset_contract_address: contractDetailed.address,
+            limit: '50',
+            offset: String(offset * 50),
+        }),
```

Comment:
```suggestion
        urlcat(apiURL, {
            owner,
            asset_contract_address: contractDetailed.address,
            limit,
            offset: offset * limit,
        }),
```

plz extract result as variable, and add `limit` variable.